### ISSUE TITLE
Tokenizer dies on buffer underrun

### ISSUE THREAD

**Issue by mikolalysenko** [Floby/node-tokenizer#2]
For example:

```
 var t = new Tokenizer();
 t.end();
```

Will crash the library with:

```
Users/mikolalysenko/GitHub/ppm/node_modules/tokenizer/lib/Tokenizer.js:68
        this._gotToken(data.substr(last_tokenized), matching[0]);
                                                            ^
TypeError: Cannot read property '0' of undefined
    at Tokenizer._tokenize (/Users/mikolalysenko/GitHub/ppm/node_modules/tokenizer/lib/Tokenizer.js:68:61)
    at Tokenizer.write (/Users/mikolalysenko/GitHub/ppm/node_modules/tokenizer/lib/Tokenizer.js:20:10)
    at Tokenizer.end (/Users/mikolalysenko/GitHub/ppm/node_modules/tokenizer/lib/Tokenizer.js:77:10)
    at Object.<anonymous> (/Users/mikolalysenko/GitHub/ppm/node_modules/tokenizer/examples/underrun.js:4:3)
    at Module._compile (module.js:449:26)
    at Object.Module._extensions..js (module.js:467:10)
    at Module.load (module.js:356:32)
    at Function.Module._load (module.js:312:12)
    at Module.runMain (module.js:492:10)
    at process.startup.processNextTick.process._tickCallback (node.js:244:9)
```

**Comment by Floby** [comment]
I pushed some changes at 702d835 and published it as 0.1.2. does it fix your problem ?

**Comment by mikolalysenko** [comment]
Yep!
